src/sentry/lang/native/appconnect.py

```diff
@@ -383,4 +383,13 @@ def list_builds(self, kind: BuildKind = BuildKind.ALL) -> List[BuildInfo]:
                         build_number=build["version"],
                     )
                     builds.append(build)
+
+        def _try_int(x: str) -> int:
+            try:
+                return int(x)
+            except ValueError:
+                return 0
+
+        builds.sort(key=lambda x: _try_int(x.build_number), reverse=True)
```

Comment:
For future souls, please describe why this is sorted.  The API does treat these `build_number`s as opaque strings (it calls them `version`) so this is a bit of a stretch.

I think I'd prefer to make `BuildInfo` ordered instead and leave it up to the caller to sort this list.  That puts the sorting logic in the download task where it belongs.

Reply:
You can also check whether there's a way to sort this via the API request. I've seen this in the ASC web app, although I didn't check if we're using the same API or not.

Reply:
The `iris` (= web-facing API) has a `sort=-version` param for the `builds` endpoint. Though we are not using that endpoint here, so not sure if the same params are available for the officially documented API.